### PR TITLE
Add era5 2008 surface fluxes

### PR DESCRIPTION
**Pull request by imreddyTeja**
Checklist:
- [x] I created a new folder `$artifact_name`
  - [x] I added a `README.md` in that that folder that
    - [x] describes the data and processing done to it
    - [x] lists the sources of the raw data
    - [x] lists the required citation, licenses
  - [x] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [x] I added the scripts that retrieve, process, and produce the artifact
  - [x] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [x] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [x] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [x] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [x] I added a link to the main `README.md` to point to the new artifact

Example plots:

mslhf_map
msror_map
msshf_map
mssror_map
msuwlwrf_map
msuwswrf_map

@AlexisRenchon @kmdeck Do these plots seem reasonable? 
Also, would you prefer the averages by hour of day and monthly averages to be in the same artifact, or separate artifacts?

### CODE REVIEW

**Comment by AlexisRenchon** [comment]
Thank you Teja!

They look good to me, and for average by hour of day and monthly averages, they should be separate artifacts.

@kmdeck , do you agree?

**Comment by kmdeck** [comment]
> Thank you Teja!
> 
> They look good to me, and for average by hour of day and monthly averages, they should be separate artifacts.
> 
> @kmdeck , do you agree?

Im confused! Calibration data is OK to be monthly only. The forcing data is needed hourly.  This PR is for calibration data.

**Comment by kmdeck** [comment]
It would also be great to have the runoff data. Is this PR meant to close Issue 903?
https://github.com/CliMA/ClimaLand.jl/issues/903

**Comment by AlexisRenchon** [comment]
> > Thank you Teja!
> > They look good to me, and for average by hour of day and monthly averages, they should be separate artifacts.
> > @kmdeck , do you agree?
> 
> Im confused! Calibration data is OK to be monthly only. The forcing data is needed hourly. This PR is for calibration data.

I believe there is a dataset in ERA5 that is average for each hour for each month? 
Which could be helpful to e.g., calibrate diurnal patterns. 

It is not a priority but could be helpful at some point.

**Comment by imreddyTeja** [comment]
> > > Thank you Teja!
> > > They look good to me, and for average by hour of day and monthly averages, they should be separate artifacts.
> > > @kmdeck , do you agree?
> > 
> > 
> > Im confused! Calibration data is OK to be monthly only. The forcing data is needed hourly. This PR is for calibration data.
> 
> I believe there is a dataset in ERA5 that is average for each hour for each month? Which could be helpful to e.g., calibrate diurnal patterns.
> 
> It is not a priority but could be helpful at some point.

Yes this is correct. By "average by hour of day", I meant average for each hour for each month. Sorry for the confusion. This PR is meant to close #903, and I am adding runoff now.

File: era5_monthly_averages_2008/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## Downloaded dataset
+
+The downloaded dataset contains the following variables:
+
+1. `mean_surface_downward_long_wave_radiation_flux`
+2. `mean_surface_downward_short_wave_radiation_flux`
+3. `mean_surface_latent_heat_flux`
+4. `mean_surface_net_long_wave_radiation_flux`
+5. `mean_surface_net_short_wave_radiation_flux`
+6. `mean_surface_sensible_heat_flu`
```
Comment (kmdeck) [inline era5_monthly_averages_2008/README.md:31]: flu -> flux

and can add runoff - thanks!

File: era5_monthly_averages_2008/README.md
Comment (kmdeck) [inline era5_monthly_averages_2008/README.md:1]: this is very clear, thank you!

File: era5_monthly_averages_2008/subset_of_data.jl [hunk 2]
```diff
@@ -0,0 +1,109 @@
+using NCDatasets
+using Dates
+
+function create_new_ds_from_time_indx(
```
Comment (kmdeck) [inline era5_monthly_averages_2008/subset_of_data.jl:4]: Could you add a docstring for this describing at a high level what it does?

**Review by kmdeck — APPROVED** [review]
Looks great, thank you! 

Could you update the issue #903 and note that you also got the monthly averaged data by hour of day also?

Could you then updated the corresponding issue for the same data, but 40 years, and make sure the same data is obtained (monthly and monthly by hour)?

**Comment by imreddyTeja** [comment]
> Looks great, thank you!
> 
> Could you update the issue #903 and note that you also got the monthly averaged data by hour of day also?
> 
> Could you then updated the corresponding issue for the same data, but 40 years, and make sure the same data is obtained (monthly and monthly by hour)?

Done. Just to keep you in the loop, I am planning on putting off the 40 years (monthly by hour) until the 3d calibration data for atmos finishes downloading. I already have the 40 years (monthly) and am processing and documenting it now. Unfortunately, the (monthly by hour) data is a much larger download. Because it is a lower priority, I thought I should get the 3d atmos data done first. The download rate is completely unpredictable, so I am not sure when that will finish. In the mean time, I'll still have the monthly data stored on sampo and create an artifact for it (although it is less than 500mb, so it could fit on central)

**Pull request merged**